aspects/schema.go

```diff
@@ -414,6 +422,8 @@ func (v *mapSchema) parseMapKeyType(raw json.RawMessage) (Schema, error) {
 	return nil, fmt.Errorf(`keys must be based on string but got %q`, typ)
 }
 
+func (v *mapSchema) mustConstrain() bool { return false }
```

Comment:
is it clear in the spec that is the case? because if that's the case it would be true also for array, no? maybe we need to decide one way or another for both. Or maybe it makes sense to do things differently for maps, so that the "map" means any-good-key -> any,  while for array one is expected to be explicit?

We should probably make these aspects clearer in the spec

Reply:
This was wrong, the spec states that maps must have constraints (the top level is manually enforced so the tests didn't catch it). I've opened https://github.com/snapcore/snapd/pull/13346 to split this logic out of this PR. It introduces a distinction between types that must have constraints (and consequently map-like definitions) and types that don't (definition can be a string). I'll rebase this PR (the arrays one) on top of that once that's stable and mark arrays to also require constraints.

Reply:
That PR has landed and I've merged master